src/tool/confiscan.sh

```diff
@@ -115,6 +122,85 @@ else
     mkdir -p "${output_dir}"
 fi
 
+###########
+# Network #
+###########
+
+[[ -d "${output_dir}/network" ]] || mkdir -p "${output_dir}/network"
+
+info "Network devices:"
+
+printf 'Device,IPv4,Netmask,Broadcast,IPv6,Prefix,DHCP,DHCP_server_identifier\n' > \
+    "${output_dir}/network/network.csv"
+while read -r device; do
+    inet4="$(ifconfig "${device}" | awk '/inet / {print $2,$4,$6}' | sed 's/ ,/,/g;s/,,/,/g')"
+    inet6="$(ifconfig "${device}" | awk '/inet6 / {print $2,$4}' | sed 's/ ,/,/g;s/,,/,/g')"
+
+    ipv4="$(printf '%s' "${inet4}" | cut -d' ' -f1)"
+    netmask="$(printf '%s' "${inet4}" | cut -d' ' -f2)"
+    broadcast="$(printf '%s' "${inet4}" | cut -d' ' -f3)"
+    ipv6="$(printf '%s' "${inet6}" | cut -d' ' -f1)"
+    prefix="$(printf '%s' "${inet6}" | cut -d' ' -f2)"
+
+    # || : is used to prevent the script from exiting when no lease file is found.
+    dhcp="$(grep 'dhcp-server-identifier' \
+        "/var/lib/dhcp/dhclient.${device}.leases" 2> /dev/null
+    )" || :
+    dhcp_server_identifier="$(grep 'dhcp-server-identifier' \
+        "/var/lib/dhcp/dhclient.${device}.leases" 2> /dev/null | \
+        awk '{print $3}' | \
+        tail -n1 | \
+        sed 's/;//g'
+    )" || :
+
+    [[ -z "${ipv4}" ]] && ipv4="N/A"
+    [[ -z "${netmask}" ]] && netmask="N/A"
+    [[ -z "${broadcast}" ]] && broadcast="N/A"
+    [[ -z "${ipv6}" ]] && ipv6="N/A"
+    [[ -z "${prefix}" ]] && prefix="N/A"
+    [[ -z "${dhcp_server_identifier}" ]] && dhcp_server_identifier="N/A"
+    if [[ -z "${dhcp}" ]]; then
+        dhcp="false"
+    else
+        dhcp="true"
+    fi
+
+    printf '%s,%s,%s,%s,%s,%s,%s,%s\n' \
+        "${device}" \
+        "${ipv4}" \
+        "${netmask}" \
+        "${broadcast}" \
+        "${ipv6}" \
+        "${prefix}" \
+        "${dhcp}" \
+        "${dhcp_server_identifier}" >> \
+        "${output_dir}/network/network.csv"
+done < <(ip -o link show | awk -F': ' '{print $2}')
```

Comment:
Something tells me this can be done much more efficiently

Reply:
Refactored in commit: https://github.com/AntonVanAssche/hogent-bachelorproef/pull/22/commits/87344c8ea0477e1647cf2b34b83f9653f180d074.

Reply:
I think you did a great job here to be honest.